src/js/administration/querys.ts

```diff
@@ -2,12 +2,24 @@ import { useMutation, useQuery, useQueryClient } from "react-query";
 import { Request } from "../app/request";
 import { AdministratorRoles } from "./types";
 
+export const roleKeys = {
+    all: ["roles"] as const,
+};
+
 const getAdministratorRoles = () => Request.get("/admin/roles").then(response => response.body);
 
 export const useGetAdministratorRoles = () => {
     return useQuery("roles", getAdministratorRoles);
 };
 
+export const userKeys = {
+    all: ["users"] as const,
+    lists: () => [...userKeys.all, "list"] as const,
+    list: (filters: Array<string | number | boolean>) => [...userKeys.lists(), ...filters] as const,
+    details: () => [...userKeys.all, "details"] as const,
+    detail: (user_id: string) => [...userKeys.details(), user_id] as const,
```

Comment:
It is difficult to understand what is going on here at a glance.

I'm not sure of the benefit of these `somethingKeys` objects. It seems that each property is only read in a single hook (eg. `useGetUsers`) and that the hook is the best place to define the keys once.

Thoughts?